lib/execution_engine2/utils/arg_processing.py

```diff
@@ -0,0 +1,31 @@
+"""
+Functions for processing arguments / parameters, including argument validity checkers and
+normalizers.
+"""
+
+from typing import Union
+
+
+def parse_bool(putative_bool: Union[str, bool, int, None]) -> bool:
+    """
+    Parse a string, bool, or int to a boolean value.
+    Strings containing 'true' or 'false', regardless of capitalization, are considered booleans.
+
+    Raises ValueError if the value cannot be parsed.
+    """
+    if putative_bool is None:
+        return False
+
+    if isinstance(putative_bool, bool):
+        return putative_bool
+
+    if isinstance(putative_bool, int):
+        return bool(putative_bool)
+
+    if isinstance(putative_bool, str):
```

Comment:
Do we care about cases when it's a string not equal to "true" or "false" ? like "tru"?

Reply:
Ah never mind it throws the exception.. What about "1.0", "1", which is likely to come from the json rpc endpoints?

Reply:
Added handling for stringified ints and floats